src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -918,6 +918,104 @@ class OPENASSETIO_CORE_EXPORT Manager {
                             const BatchElementErrorCallback& errorCallback,
                             const trait::TraitSet& resultTraitSet = {});
 
+  /**
+   * Callback signature used for a successful paged entity relationship
+   * query.
+   */
+  using PagedRelationshipSuccessCallback =
+      std::function<void(std::size_t, EntityReferencePagerPtr)>;
+
+  /**
+   * Paged version of getWithRelationship. See non-paged
+   * @ref getWithRelationship for further documentation.
+   *
+   * @param relationshipTraitsData The traits of the relationship to
+   * query.
+   *
+   * @param entityReferences A list of @ref entity_reference to query
+   * the specified relationship for.
+   *
+   * @param pageSize The size of each page of data. The page size is
+   * fixed for the lifetime of pager object given to the @p
+   * successCallback.
```

Comment:
[question] Should we define this should be > 0 and/or introduce any validation?

Reply:
:thinking: `size_t` is unsigned, so negative values are not a problem.  But `== 0` is still an issue.

I guess we should validate, since zero is potentially quite bad, given that some implementations may well use the page size in arithmetic (i.e. division by zero).

Reply:
We have (or maybe had) prior art where `Manager` used to validate array lengths etc. when multiple args needed to be coherent.

Reply:
Added a an index error for when the size is zero.
https://github.com/OpenAssetIO/OpenAssetIO/pull/986/commits/29e07320989366f23bb3886f525e8e437bee2dec

and test : 
https://github.com/OpenAssetIO/OpenAssetIO/pull/986/commits/db6f4c2e9f4548f33365d9f68cfc440547a75ba1